### ISSUE TITLE
Converging of the network for different data sets?

### ISSUE THREAD

**Issue by shamanez** [snowkylin/ntm#12]
I tried to train this network to identify 13 classes with 5 RGB images for each class. 
One image is like this.

![20171221_175923](https://user-images.githubusercontent.com/16892570/34347682-46c5bb1a-ea40-11e7-92dc-37e0891beb73.jpg)

I modified the network to work with RGB. But even after 100000 iterations cannot see any kind of convergence. 

Do you think this network is not capable of remembering information in above-mentioned images? Because in character data set the information is not complex as much as in above type images.

**Comment by snowkylin** [comment]
I am not very sure about this, but generally NTM is harder to converge compared with other DL models, which is a main drawback of this model.

**Comment by albertchristianto** [comment]
Hi @snowkylin, I also faced some problem with RGB images. Have you tried MANN implementation for RGB images. 
I have tried your code and this MANN code (https://github.com/hmishra2250/NTM-One-Shot-TF) for RGB images. hmishra2250's code converged but it takes a long time. 
On the other hand, your code even can't converged. 
I am just curious what is the different between your code and hmishra2250's code? 
I have tried to compare your code and hmishra2250's code. But, because I am still new on this topic, I haven't (maybe couldn't) found any useful information. Thanks in advanced.
Best regards,
Albert

**Comment by snowkylin** [comment]
@albertchristianto No, I haven't tried RGB images. Maybe you can try to convert images into gray scale and see if there is any change on the performance of the model. Sorry that I am busy these days and cannot afford much time to analyse other codes.

**Comment by albertchristianto** [comment]
@snowkylin I extracted the feature using VGG16 then use it as my MANN input. it's okay. Thank you for your quick response.
@shamanez for the hmishra2250's code converged (also the accuracy) but it takes a long time. but hmishra2250's code didn't have very clear explanation about what happen in the training process, therefore i am kind of doubting the result. I also have a hard time to modify the parameter from hmishra2250's code.
